### PR TITLE
Rest Client: Add property to skip hostname verification

### LINKED ISSUE #27901: RestClient Reactive -  set HostnameVerifier or SSLContext
### Description

There is a [known limitation](https://quarkus.io/guides/rest-client-reactive#known-limitations) on RestClient Reactive, we cannot set a HostnameVerifier or SSLContext. 

We have switched back to rest-client-mutiny for now, even if it's deprecated, but it would be useful to get this done, so we can move back towards a fully reactive service.

Do we know the priority of this feature? At least for HostnameVerifier.

Context: It is very common to give user, when configuring webhooks, the option to use insecure SSL at their own expense, usually for testing purposes, and we want to give that to our users.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by Sgitario**
Before these changes, we only can disable the hostname verification in Rest Client classic by providing the following property:

```
quarkus.rest-client.extensions-api.hostname-verifier=io.quarkus.restclient.NoopHostnameVerifier
```

However, this is not working in Rest Client reactive because setting a hostname verifier strategy is not supported by the Vert-x HTTP Client. 

With these changes, we have added a new property in both Rest Client classic and reactive `quarkus.rest-client.extensions-api.verify-host=true or false`. In Rest Client classic, when disabling the verify host, internally it will add the `NoopHostnameVerifier` strategy. In Rest Client reactive, it will properly configure the Vert.x HTTP client to disable the hostname verification. Therefore, in both Rest Client implementations (classic and reactive), the behaviour is the same. 

Fix https://github.com/quarkusio/quarkus/issues/27901

### CODE REVIEW

**Comment by geoand** [comment]
@famod mind trying this out?

**Comment by famod** [comment]
@geoand I'd love to, but right now I'm standing in the middle of chaos because on the weekend we're finally relocating to our new house.
So better not wait for me.

**Comment by geoand** [comment]
Hope all goes well with the move :)

**Comment by geoand** [comment]
@Chexpir would you like to test this before we merge it?

**Review by gsmet — DISMISSED** [review]
Asked an important question :).

File: docs/src/main/asciidoc/rest-client-reactive.adoc
Comment (gsmet) [reply docs/src/main/asciidoc/rest-client-reactive.adoc:247]: Asking that because somebody asked with the behavior of `quarkus.tls.trust-all` in conjunction with REST Clients. That being said, being able to adjust things specifically for REST Client would be useful anyway so I think this patch has value.

Comment (cescoffier) [reply docs/src/main/asciidoc/rest-client-reactive.adoc:247]: Yes, `quarkus.tls.trust-all` is global. Having a rest client attribute would be nice. However, I would keep the name `trust-all`.

Comment (famod) [reply docs/src/main/asciidoc/rest-client-reactive.adoc:247]: Are you talking about a (per-client) `trust-all` that skips more verifications _instead_ of `verify- host` that skips only the hostname verification?
Or is this just a naming discussion?

Comment (cescoffier) [reply docs/src/main/asciidoc/rest-client-reactive.adoc:247]: Trust-all by-pass everything (basically pass a ssl context that accept any certs).

Comment (famod) [reply docs/src/main/asciidoc/rest-client-reactive.adoc:247]: Yes, that's why I'm asking. `trust-all` (global or per client) is a sledgehammer which is usually unacceptable in production (where you might have cases though in which you can only use an IP to talk to another system).
So my point is that `trust-all` (on whichever level) should not "replace" `verify-host`.

So, on client level, `trust-all` _additionally_ to `verify-host` would be ok for me, but having _only_ `trust-all` is much too coarse.

**Comment by Chexpir** [comment]
@geoand thanks for the proposition, no need to test it. I've reviewed the code and I trust that it will work!
Anyway, it will be useful to know the process to build and test github branches, it might become handy in the future, as we are into Quarkus for the long term! (I have searched and could not find anything).

**Comment by geoand** [comment]
I would say that the easiest way to checkout a Pull Request is following [this](https://cli.github.com/manual/gh_pr_checkout).
Once that is done, you can build it locally by following [this](https://github.com/quarkusio/quarkus/blob/main/CONTRIBUTING.md#build).

File: docs/src/main/asciidoc/rest-client-reactive.adoc
Comment (geoand) [reply docs/src/main/asciidoc/rest-client-reactive.adoc:247]: I agree with @famod on this.

**Comment by geoand** [comment]
@Chexpir were you able to test this one?

**Comment by geoand** [comment]
Were are we with this one?

I personally think we should get it in

**Comment by Sgitario** [comment]
> Were are we with this one?
> 
> I personally think we should get it in

My +100 to get it in.

**Comment by cescoffier** [comment]
My last comment was to make sure we clearly say in the doc that skipping host name verification should not be used in prod ('use it to your own risk')

**Comment by geoand** [comment]
We should 100% add that

**Comment by Sgitario** [comment]
PR updated with adding this warning to the docs.

**Comment by Sgitario** [comment]
Any updates on this pull request? I would like to proceed to merge it if everybody agrees.

**Review by cescoffier — APPROVED** [review]

**Comment by fasuke85** [comment]
Hi, it would be awsome if you merged this PR 😄

**Comment by oven** [comment]
This issue is a blocker for us because we have to communicate with an HTTPS service over an ssh tunnel. Please merge :)

**Comment by geoand** [comment]
@gsmet any last words or should I dismiss the old review?

**Comment by geoand** [comment]
Since the last CI run of this was a while ago, I rebased onto `main`.

**Pull request merged**
